sdk/core/core-http/lib/atomError.ts

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { Constants } from "./util/constants";
+import { HttpOperationResponse } from "./httpOperationResponse";
+
+/**
+ * Represents custom `Error` information returned by ATOM based services.
+ */
+export class AtomError extends Error {
+  code?: string;
+  statusCode?: number;
+  additionalProperties?: any;
+  requestId?: string;
+
+  constructor(message: string) {
+    super(message);
+    this.additionalProperties = {};
+    Object.setPrototypeOf(this, AtomError.prototype);
+  }
+}
+
+/**
+ *
+ * Utility to help construct the normalized `AtomError` object based on given `errorBody`
+ * data and other data present in the received `response` object.
+ *
+ * @param errorBody
+ * @param response
+ */
+export function buildAtomError(errorBody: any, response: HttpOperationResponse): AtomError {
+  if (typeof errorBody === "string") {
+    return new Error(errorBody);
```

Comment:
Is there a reason this isn't also an `AtomError`? It seems like you could populate the `statusCode` and `requestId` fields from the `response` if they are available.

Reply:
I agree with @chradek. Even when `errorBody` is a string, we should try and use the code/status from the response

Reply:
This one's updated in #4922